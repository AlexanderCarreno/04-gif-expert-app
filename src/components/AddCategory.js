import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({
  setCategories
}) => {

const [inputValue, setInputValue] = useState('');

const handleInputChange = (e)=> {
    //console.log(e.target.value);
    setInputValue( e.target.value );

    //console.log('handleInputChanges llamado');
}

const handleSubtim = (e) => {
    e.preventDefault();
    //console.log('handleSubmit', inputValue);

    if( inputValue.trim().length > 2 ){
      setCategories( cats => [inputValue, ...cats]);
      setInputValue('');
      //console.log('Submit hecho');
    }
}

  return (
    <>
    <h1>{inputValue}</h1>
    <form onSubmit={ handleSubtim }>
      <p>{ inputValue }</p>
        <input
            type="text"
            value= { inputValue }
            onChange={ handleInputChange }
        />
    </form>
    </>
  )
}


AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}