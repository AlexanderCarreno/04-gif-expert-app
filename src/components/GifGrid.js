import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ( {category} ) => {
  

  //const [images, setImages] = useState([]);
  //useEffect( () => {
  //  getGifs( category )
  //    .then( imgs => {
  //      //setImages( imgs );
  //      setImages
  //    });
  //}, [ category ]);



  const { data:images, loading } = useFetchGifs( category );


/*
  return (
    <>
      <h3>GifGrid</h3>
      <ol>
        {
          images.map( img => ( 
            <li key={ img.id }>{ img.title }</li>
          ))
        }
      </ol>
    </>
  )
  */

  return (
    <>
      <h3 className='animate__animated animate__fadeIn animate__delay-1s'>{ category }</h3>
      { /*loading ? <p>Loading</p> : null*/ }
      { loading && <p className='animate__animated animate__flash'>Loading</p> }

      <div className='card-grid'>
        {
          images.map( img => ( 
            <GifGridItem 
              key={ img.id }
              { ...img }
            />
          ))
        }
      </div>
    
      {/*

      <div className='card-grid'>
        <ol>
          {
            images.map( img => ( 
              <GifGridItem 
                key={ img.id }
                { ...img }
              />
            ))
          }
        </ol>
      </div>*/
      }
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}