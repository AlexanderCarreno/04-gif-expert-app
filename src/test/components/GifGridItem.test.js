import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('debe mostrar el componente correctamente', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('debe mostrar <GitExpertApp /> corractamente', () => { 
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el titulo', () => { 
         const p = wrapper.find('p');
         expect( p.text().trim() ).toBe( title );
    });

    test('debe de tener la imagen igual al url y alt de los props', ()=>{
        const img = wrapper.find('img');
        //console.log(img.html());
        //console.log(img.prop('src'));
        expect( img.prop('src')).toBe(url);
        expect( img.prop('alt')).toBe(title);
    });

    test('debe tener la clase animate__fadeIn', () => { 
        const div = wrapper.find('div');
        //console.log(div.prop('className'));
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn')).toBe(true);
        //expect( className.includes('animate__fadeIn')).not.toBe(true);
     })
});