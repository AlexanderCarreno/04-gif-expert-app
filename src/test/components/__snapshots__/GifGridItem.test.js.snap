// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar el componente correctamente debe mostrar <GitExpertApp /> corractamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
